.github/workflows/bin/generate_cron_workflows.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+
+import sys
+from jinja2 import Environment, FileSystemLoader
+
+
+def validate_inputs(cli_args):
+    try:
+        epoch = cli_args[1]
+    except Exception:
+        raise ValueError('Missing required parameter. '
+                         'Epoch must be provided.')
+
+    return epoch
+
+
+if __name__ == '__main__':
```

Comment:
Thanks @thermokarst! This was the original approach that I had taken, and was what I had understood as the idiomatic approach - I added the function definition under the `if __name__ == '__main__'` guard because of [this comment](https://github.com/qiime2/package-integration/pull/70#discussion_r749747803) from your earlier review, but I must have misunderstood your request there. Could you clarify what you were referring to in that comment?

Reply:
```python
def foo(arg1):
    print(arg1)
    # do stuff with arg1
    

if __name__ == '__main__':
    foo('oh hello')
```

vs

```python
if __name__ == '__main__':
    val = 'oh hello'
    print(val)
    # do stuff with val
```

The second code block above is an example of elision, you're getting rid of the standalone function and just putting the code "in-line" where it is used. Taking a step back, a common reason we often package up code in a function is for re-use (or unit testing) - in this case I can't see much need for reuse, though, which was why I originally suggested eliding. Then in your response, you moved the whole function into the guard, which is probably going to be confusing to future readers, rather than suggesting eliding again I figured just keeping it as a standalone function is fine, in which case it should be defined outside of the guard.

Moving forward, I have no preference, but I think it should be one or the other (elided, or a standalone function defined in the main body of the script).